declarative-pipeline-migration-assistant/src/main/java/io/jenkins/plugins/todeclarative/converter/freestyle/FreestyleToDeclarativeConverter.java

```diff
@@ -112,40 +110,92 @@ public String toGroovy()
+        Map<JobPropertyDescriptor, JobProperty<? super FreeStyleProject>> projectProperties = freeStyleProject.getProperties();
+        // ignore JiraProjectProperty as it is not removed when removing jira publishers so we avoid false positive
+        projectProperties.entrySet().removeIf(entry -> entry.getKey().clazz.getName().equals("hudson.plugins.jira.JiraProjectProperty"));
+        convertTypeList(converterRequest, converterResult, converterList, projectProperties.values());
+        convertTypeList(converterRequest, converterResult, converterList, freeStyleProject.getBuilders());
+        convertTypeList(converterRequest, converterResult, converterList, freeStyleProject.getPublishersList());
+    }
 
+    protected <T> void convertTypeList(ConverterRequest request, ConverterResult result, ExtensionList<BaseConverter> converterList, Collection<T> typeList) throws ConverterException {
+        ModelASTStages stages = result.getModelASTPipelineDef().getStages();
+        if ( stages == null )
+        {
+            stages = new ModelASTStages( this );
+            result.getModelASTPipelineDef().setStages( stages );
+        }
+        for ( T type : typeList )
+        {
+            boolean matched = false;
+            for ( BaseConverter converter : converterList )
+            {
+                if ( converter.canConvert(type) )
+                {
+                    matched = true;
+                    if ( converter.convert(request, result, type) )
+                    {
+                        result.addConvertedType(type.getClass().getName());
+                    } else
+                    {
+                        // TODO: log addition information for failed conversion?
+                    }
+                    if ( type instanceof Maven ) {
+                        // Maven is a special one and we can apply only one converter so we pick the first one
+                        break;
+                    }
+                }
+            }
+            if ( !matched )
+            {
+                result.addWarning( new Warning( "Converter not found '" + getDisplayName((Describable) type) + "'", type.getClass() ) );
+                SingleTypedConverter noConverter = null;
+                if ( type instanceof Builder )
+                {
+                    noConverter = Jenkins.get().getExtensionList( NoBuilderConverter.class ).get(0);
+                } else if ( type instanceof Publisher )
+                {
+                    noConverter = Jenkins.get().getExtensionList( NoPublisherConverter.class ).get(0);
+                }
+                if ( noConverter != null )
+                {
+                    noConverter.convert( request, result,type );
+                }
+            }
+        }
     }
 
     protected void convertBuildTriggers( ConverterRequest converterRequest, ConverterResult converterResult,
```

Comment:
If the refactoring is accepted, we can probably remove all the convertXXX methods below

Reply:
removed in d469cac